onmt/inputters/text_dataset.py

```diff
@@ -5,31 +5,14 @@
 from torchtext.data import Field, RawField
 
 from onmt.inputters.dataset_base import DatasetBase
+from onmt.inputters.datareader_base import DataReaderBase
 
 
-class TextDataset(DatasetBase):
-    """
-    Build `Example` objects, `Field` objects, and filter_pred function
-    from text corpus.
-
-    Args:
-        fields (dict): a dictionary of `torchtext.data.Field`.
-            Keys are like 'src', 'tgt', 'src_map', and 'alignment'.
-        src_examples_iter (dict iter): preprocessed source example
-            dictionary iterator.
-        tgt_examples_iter (dict iter): preprocessed target example
-            dictionary iterator.
-        dynamic_dict (bool)
-    """
-
-    @staticmethod
-    def sort_key(ex):
-        if hasattr(ex, "tgt"):
-            return len(ex.src[0]), len(ex.tgt[0])
-        return len(ex.src[0])
+class TextDataReader(DataReaderBase):
+    def __init__(self, **kwargs):
```

Comment:
This `__init__` is unnecessary.

Reply:
I've designed it so that any reader can take the union of all the reader args, if that makes sense. So the ``**kwargs`` is doing something. That way in translator.py ``Translator.__init__`` and in preprocess.py ``main`` you don't have to do logic on the datatype to decide what arguments to pass in to the reader. You just pass them all in and whatever subclass takes what it wants and ignores the rest.

I would say that should be in the ``DataReaderBase`` "contract." However, putting the empty ``__init__`` in the base class suggests that subclasses need to call super init, and that's not the case.

Do you have something else in mind?

Reply:
I understand the reasoning but it sounds like it would make more sense on the `DataReaderBase` -- otherwise, additional data readers will need to define this dummy constructor as well. Could you add a comment so no one makes the same mistake as I did?

Reply:
Done.